### ISSUE TITLE
ContextImpl not correctly attached to Application during ParallelUniverse.setUpApplicationState()

### ISSUE THREAD

**Issue by andycmaj** [robolectric/robolectric#598]
### Some context first:

I am trying to write to the temp directory (ShadowContext.FILES_DIR) that gets assigned to ApplicationInfo.dataDir by getting context.getApplicationInfo().dataDir.

When RobolectricPackageManager (called by ParallelUniverse.setUpApplicationState()) creates the ApplicationInfo and assigns **applicationInfo.dataDir = ShadowContext.FILES_DIR.getAbsolutePath()**, dataDir has a value. This value is reflected in **contextImpl** created reflectively via **createPackageContext[line 108]. 
### Issue:

After ParallelUniverse.setUpApplicationState[line 114] reflectively calls application.attach(contextImpl), effectively wrapping **contextImpl** with the Application instance, the ApplicationInfo attached to **contextImpl** seems to get lost or is no longer accessible via the Application (which is wrapping contextImpl).

The correct ApplicationInfo instance is still attached to the **contextImpl** instance (contextImpl.getApplicationInfo().dataDir is correct), but is not available when calling application.getApplicationInfo() (application.getApplicationInfo().dataDir, which should delegate to contextImpl, is null).
### TLDR:

**Basically, ApplicationInfo is lost when ParallelUniverse sets up and attaches Context to an existing Application instance.**
### Failed debugging attempt:

Looks like when Application.attach() is called, it does the actual attaching by calling Application.attachBaseContext (which is a method of Application's superclass, ContextWrapper). I experienced weird behavior when stepping through Application.attach in a debugger: the call from attach() to [super.]attachBaseContext seems to be skipped or cannot be stepped into in the debugger, possibly having to do with the fact that Application.attach() seems to be proxied to something that looks like $$robo$$Application_0b64_attach().
